pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -73,6 +78,7 @@ func newLeaderEngine() *LeaderEngine {
 // It is ONLY to be used for tests
 func ResetGlobalLeaderEngine() {
 	globalLeaderEngine = nil
+	telemetry.Reset()
```

Comment:
I dont think it is the right place to do the `Reset()`. I feel that the `telemetry.Reset()` should be done in the test and not hidden in this function. WDYT ?